src/crimson/osd/main.cc

```diff
@@ -183,6 +183,22 @@ seastar::future<> fetch_config()
   });
 }
 
+static void override_seastar_opts(std::vector<const char*>& args)
+{
+#ifndef CRIMSON_M2N
+  if (auto found = std::find_if(std::begin(args), std::begin(args),
```

Comment:
Switched the algorithm per @ronen-fr's advice. Now we just check somebody already passed `--smp` instead of matching `argv[0]` to `ceph-osd`.

Reply:
Remove the ifndef.  It's irrelevant for now, and I really don't think we'll be using a compile flag anyway

Also, please don't use m2n, m-to-n, or any similar variant when referring to multi-core support.

Reply:
I think we should support multi-core such as seastar automatically without use compile flag.

Reply:
Part of actually supporting it will be adding a default mode where it detects cpus and adjusts automatically as well as a command line param to specify the number of cpus.  For now, simply adding --smp 1 is fine since we cannot make use of more than that anyway.

Reply:
@athanatos the M2N thing is just for grepping so we can find all the places where we should fix for multi core support.

Reply:
Perhaps let's go with MULTI_CORE in a comment instead?